src/rt/sections_elf_shared.d

```diff
@@ -711,12 +711,12 @@ void scanSegments(in ref dl_phdr_info info, DSO* pdso)
         case PT_LOAD:
             if (phdr.p_flags & PF_W) // writeable data segment
             {
-                auto beg = cast(void*)(info.dlpi_addr + phdr.p_vaddr);
+                auto beg = cast(void*)(info.dlpi_addr + (phdr.p_vaddr & ~(size_t.sizeof - 1)));
                 pdso._gcRanges.insertBack(beg[0 .. phdr.p_memsz]);
             }
             version (Shared) if (phdr.p_flags & PF_X) // code segment
             {
-                auto beg = cast(void*)(info.dlpi_addr + phdr.p_vaddr);
+                auto beg = cast(void*)(info.dlpi_addr + (phdr.p_vaddr & ~(size_t.sizeof - 1)));
                 pdso._codeSegments.insertBack(beg[0 .. phdr.p_memsz]);
```

Comment:
I'm not sure if this is always 100% correct: `phdr.p_vaddr & ~(size_t.sizeof - 1)` rounds down the address to the next smaller aligned address. If this happens `beg[0 .. phdr.p_memsz]` will in theory not include the complete section and miss up to `size_t.sizeof - 1` bytes at the end. Being pedantic this needs to be `beg[0 .. phdr.p_memsz + (phdr.p_vaddr & (size_t.sizeof - 1))]`.

Example: p_vaddr = 6, p_memsz = 4, size_t.sizeof = 4 correct range: [6..10] after rounding: [4..8], correct and aligned range : [4..10]